src/settings.rs

```diff
@@ -219,6 +219,27 @@ impl Settings {
 
     #[allow(dead_code)]
     pub fn validate_and_warn(&self) {
+        //
+        // JWT RBAC
+        //
+        // Using HMAC-SHA256, recommended secret size is 32 bytes
+        const JWT_RECOMMENDED_SECRET_LENGTH: usize = 256 / 8;
+
+        // Log if JWT RBAC is enabled but no API key is set
+        if self.service.jwt_rbac.unwrap_or_default() {
+            if self.service.api_key.clone().unwrap_or_default().is_empty() {
+                log::error!("JWT RBAC enabled but no API key set")
```

Comment:
I'd prefer to demote this to a `warn` instead, they both have the same visibility in logs anyway.

```suggestion
                log::warn!("JWT RBAC enabled but no API key set")
```

Reply:
Ok, the reason I used error here is that the requested functionality (enabling JWT RBAC) is not applied so users might complain that something is not working but there are no errors in the log.

Reply:
Updated to `warn` and changed wording a bit to make it more clear to the user.